cmd/ipfs/init.go

```diff
@@ -132,23 +133,24 @@ func addDefaultAssets(out io.Writer, repoRoot string) error {
 	}
 	defer nd.Close()
 
-	dirb := uio.NewDirectory(nd.DAG)
+	dir := &merkledag.Node{Data: unixfs.FolderPBData()}
```

Comment:
On Wed, Apr 29, 2015 at 01:27:43PM -0700, Juan Batiz-Benet wrote:
> > @@ -132,23 +133,24 @@ func addDefaultAssets(out io.Writer, repoRoot string) error {
> >  	}
> >  	defer nd.Close()
> >  
> > -	dirb := uio.NewDirectory(nd.DAG)
> > +	dir := &merkledag.Node{Data: unixfs.FolderPBData()}
> 
> Curious why not using the constructor? 

Because I wanted AddNodeLink.  Cycling from node → key string → node →
AddNodeLink (which is what AddChild does [1]) seemed wasteful and
awkward (see note in 0ce5784a, cmd/ipfs/init: Replace coreunix.Add
with importer.BuildDagFromReader, 2015-04-24).

[1]: https://github.com/ipfs/go-ipfs/blob/cc5f6bb306430d146d3e5a3f7956073c10d0112b/unixfs/io/dirbuilder.go#L30